src/scheduled_tasks.rs

```diff
@@ -356,48 +357,156 @@ async fn overwrite_deleted_posts_and_comments(pool: &mut DbPool<'_>) {
-          .await
-          .map_err(|e| error!("Failed to update site stats: {e}"))
-          .ok();
-
-        let update_community_stmt = format!("update community_aggregates ca set users_active_{} = mv.count_ from r.community_aggregates_activity('{}') mv where ca.community_id = mv.community_id_", i.1, i.0);
-        sql_query(update_community_stmt)
-          .execute(&mut conn)
-          .await
-          .map_err(|e| error!("Failed to update community stats: {e}"))
-          .ok();
-      }
-
-      info!("Done.");
+      process_site_aggregates(&mut conn, interval).await;
+      process_community_aggregates(&mut conn, interval).await;
     }
     Err(e) => {
       error!("Failed to get connection from pool: {e}");
     }
   }
 }
 
+/// Re-calculate all the active counts
+async fn all_active_counts(pool: &mut DbPool<'_>, intervals: [(&str, &str); 4]) {
+  for i in intervals {
+    active_counts(pool, i).await;
+  }
+}
+
+async fn process_site_aggregates(conn: &mut AsyncPgConnection, interval: (&str, &str)) {
+  // Select the site count result first
+  let site_activity_res = sql_query(format!(
+    "select * from r.site_aggregates_activity('{}')",
+    interval.0
+  ))
+  .get_result::<SiteActivitySelectResult>(conn)
+  .await;
+
+  match site_activity_res {
+    Ok(res) => {
+      let processed_rows = res.site_aggregates_activity;
+
+      // Update the site count
+      sql_query(format!(
+        "update site_aggregates set users_active_{} = $1 where site_id = 1",
+        interval.1,
+      ))
+      .bind::<Integer, _>(processed_rows)
+      .execute(conn)
+      .await
+      .map_err(|e| error!("Failed to update site stats: {e}"))
+      .ok();
+
+      info!(
+        "Finished site_aggregates active_{} (processed {} rows)",
+        interval.1, processed_rows
+      );
+    }
+    Err(e) => {
+      error!("Failed to fetch site activity: {e}");
+    }
+  }
+}
+
+async fn process_community_aggregates(conn: &mut AsyncPgConnection, interval: (&str, &str)) {
+  // Select the community count results into a temp table.
+  let caggs_temp_table = "community_aggregates_temp_table";
+
+  let create_table_res = sql_query(format!(
+    "CREATE TEMP TABLE {caggs_temp_table} AS SELECT * FROM r.community_aggregates_activity('{}')",
+    interval.0
+  ))
+  .execute(conn)
+  .await;
+
+  match create_table_res {
+    Ok(_) => {
+      // Split up into 1000 community transaction batches
+      let update_batch_size = 1000;
+      let mut processed_rows_count = 0;
+      let mut prev_community_id_res = Some(0);
+
+      while let Some(prev_community_id) = prev_community_id_res {
+        let result = sql_query(format!(
+          "UPDATE community_aggregates a
+            SET users_active_{} = b.count_
+            FROM (
+              SELECT count_, community_id_
+              FROM {caggs_temp_table}
+              WHERE community_id_ > $1
+              ORDER BY community_id_
+              LIMIT $2
+              FOR UPDATE SKIP LOCKED
```

Comment:
This was used in other batch updates for hot ranks, hopefully it will help with locking and skip those that are.